src/query-creator.ts

```diff
@@ -464,39 +464,57 @@ export class QueryCreator<DB> {
    * console.log(result.numUpdatedRows)
    * ```
    */
-  updateTable<TR extends keyof DB & string>(
-    table: TR,
+  updateTable<TE extends keyof DB & string>(
+    from: TE[],
   ): UpdateQueryBuilder<
     DB,
-    ExtractTableAlias<DB, TR>,
-    ExtractTableAlias<DB, TR>,
+    ExtractTableAlias<DB, TE>,
+    ExtractTableAlias<DB, TE>,
     UpdateResult
   >
 
-  updateTable<TR extends AnyAliasedTable<DB>>(
-    table: TR,
+  updateTable<TE extends TableExpression<DB, never>>(
```

Comment:
Whoops, this shouldn't be allowed. Fixing.

Reply:
Arbitrary expressions ARE okey it seems... You just naturally can't update them. Should we allow them or not? I'm leaning towards allowing them in the array but not as the lone target.